packages/aws-cdk-lib/aws-lambda/lib/function-base.ts

```diff
@@ -418,20 +431,29 @@ export abstract class FunctionBase extends Resource implements IFunction, ec2.IC
 
   /**
    * Grant the given identity permissions to invoke this Lambda
+   * @param grantee The principal (identity) to grant invocation permission.
+   * @param props onlyGrantLatestVersion (Optional) Controls whether to grant access only to latest function versions. Defaults to `false`.
+   *  - When set to `false`, both the function and functions with specific versions can be invoked.
+   *  - When set to `true`, only the function without a specific version (`$Latest`) can be invoked.
    */
-  public grantInvoke(grantee: iam.IGrantable): iam.Grant {
+  public grantInvoke(grantee: iam.IGrantable, props: GrantInvokeProps = {}): iam.Grant {
     const hash = createHash('sha256')
       .update(JSON.stringify({
         principal: grantee.grantPrincipal.toString(),
         conditions: grantee.grantPrincipal.policyFragment.conditions,
+        onlyGrantLatestVersion: props.onlyGrantLatestVersion,
       }), 'utf8')
       .digest('base64');
     const identifier = `Invoke${hash}`;
 
     // Memoize the result so subsequent grantInvoke() calls are idempotent
     let grant = this._invocationGrants[identifier];
     if (!grant) {
-      grant = this.grant(grantee, identifier, 'lambda:InvokeFunction', this.resourceArnsForGrantInvoke);
+      let resouceArns = this.resourceArnsForGrantInvoke;
+      if (props.onlyGrantLatestVersion) {
+        resouceArns = [this.functionArn];
```

Comment:
Is `functionArn` always fully qualified? If it is, what happens if `functionArn` points to a specific version, e.g. `my-function:1`, but other versions were published since? You are no longer granting the latest version, just "a" version

See the comment on the unit tests file below

Reply:
Refer to the comment above, the previous design doesn't expect a `functionArn` with version